pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2824,3 +2824,47 @@ type ConfigFileVolumeSource struct {
 	// projected items for all in one resources secrets, configmaps, and downward API
 	Projected *v1.ProjectedVolumeSource `json:"projected,omitempty" protobuf:"bytes,26,opt,name=projected"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CephCOSIDriver represents the CRD for the Ceph COSI Driver Deployment
+// +kubebuilder:resource:shortName=cephcosi
+type CephCOSIDriver struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	// Spec represents the specification of a Ceph COSI Driver
+	Spec CephCOSIDriverSpec `json:"spec"`
+}
+
+// CephCOSIDriverList represents a list of Ceph COSI Driver
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type CephCOSIDriverList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []CephCOSIDriver `json:"items"`
+}
+
+// CephCOSIDriverSpec represents the specification of a Ceph COSI Driver
+type CephCOSIDriverSpec struct {
+	// Image is the container image to run the Ceph COSI driver
+	// +optional
+	Image string `json:"image,omitempty"`
+	// ObjectProvisionerImage is the container image to run the COSI driver sidecar
+	// +optional
+	ObjectProvisionerImage string `json:"objectProvisionerImage,omitempty"`
+	// DeploymentStrategy is the strategy to use to deploy the COSI driver.
+	// Disabled: Rook will not deploy the Ceph COSI driver.
+	// Auto: Rook will deploy the Ceph COSI driver if the CSI driver is enabled.
+	// Force: Rook will deploy the Ceph COSI driver even if the CSI driver is not enabled.
+	// +optional
+	// +kubebuilder:validation:Enum=Disabled;Auto;Force
```

Comment:
Please create and use constants for these. I see several different usages in this PR: "auto" "Auto" "Force" "Forced". While thinking about this, I realize it's probably good to use terms that k8s also uses. I don't see force/disable much, but I do see Always/Never. That's probably a good choice for those familiar w/ k8s.

```
COSIDeploymentStrategyAuto = "Auto"
COSIDeploymentStrategyNever = "Never"
COSIDeploymentStrategyAlways = "Always"
```